rero_mef/marctojson/do_gnd_places.py

```diff
@@ -0,0 +1,320 @@
+        - f Konferenz
+        - g Geografikum
+        - n Person (nicht individualisiert)
+        - p Person (individualisiert)
+        - s Sachbegriff
+        - u Werk
+        """
+        for field_075 in self.marc.get_fields("075") or []:
+            if field_075.get("2") and field_075["2"] == "gndgen":
+                return RECORD_TYPES.get(field_075["b"])
+
+    def _transform(self):
+        """Call the transformation functions."""
+        record_type = self.get_type()
+        if record_type in ["bf:Place"]:
+            for func in dir(self):
+                if func.startswith("trans"):
+                    func = getattr(self, func)
+                    func()
+        else:
+            msg = f"Not a place: {record_type}"
+            if self.logger and self.verbose:
+                self.logger.warning(f"NO TRANSFORMATION: {msg}")
+            self.json_dict = {"NO TRANSFORMATION": msg}
+            self.trans_gnd_pid()
+
+    @property
+    def json(self):
+        """Json data."""
+        return self.json_dict or None
+
+    def trans_gnd_deleted(self):
+        """Transformation deleted leader 5.
+
+        $c: Redirect notification
+        $x: Redirect
+        $c: Deletion notification
+        $d: Deletion
+
+        https://www.dnb.de/EN/Professionell/Metadatendienste/Datenbezug/
+        GND_Aenderungsdienst/gndAenderungsdienst_node.html
+        """
+        if self.logger and self.verbose:
+            self.logger.info("Call Function", "trans_gnd_deleted")
+        if self.marc.leader[5] in ["c", "d", "x"]:
+            self.json_dict["deleted"] = datetime.now(timezone.utc).isoformat()
+
+    def trans_gnd_pid(self):
+        """Transformation pid from field 001."""
+        if self.logger and self.verbose:
+            self.logger.info("Call Function", "trans_gnd_pid")
+        if field_001 := self.marc.get_fields("001"):
+            self.json_dict["pid"] = field_001[0].data
+            self.json_dict["type"] = "bf:Place"
+
+    def trans_gnd_identifier(self):
+        """Transformation identifier from field 024, 035."""
+        if self.logger and self.verbose:
+            self.logger.info("Call Function", "trans_gnd_identifier")
+        fields_024 = self.marc.get_fields("024")
+        for field_024 in fields_024:
+            subfield_0 = field_024.get("0")
+            if isinstance(subfield_0, list):
+                subfield_0 = subfield_0[0]
+            subfield_2 = field_024.get("2")
+            if isinstance(subfield_2, list):
+                subfield_2 = subfield_2[0]
+            if subfield_0 and subfield_2:
+                self.json_dict.setdefault("identifiedBy", []).append(
+                    {
+                        "source": subfield_2.upper(),
+                        "type": "uri",
+                        "value": subfield_0,
+                    }
+                )
+        for field_035 in self.marc.get_fields("035"):
+            if field_035.get("a"):
```

Comment:
`if subfield_a := field_035.get("a")`?

Reply:
It is not the same. `get("a")` could be a list `["a"]` is the first element from a list.